### PR TITLE
Update mykrobe build process

### PR DESCRIPTION
**Pull request by mbhall88**
Updates the build process inline with latest version of mykrobe. Also adds some more specifics for minimum mongoengine version.

----

> Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences**
      (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

> Everyone has access to the following BiocondaBot commands, which can be given in a comment:
>
>  * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
>  * `@BiocondaBot please add label` will add the `please review & merge` label.
>  * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.
>
> For members of the Bioconda project, the following command is also available:
>
>  * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.
>
> If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

### CODE REVIEW

File: recipes/mykrobe/post-link.sh [hunk 1]
```diff
@@ -1,43 +0,0 @@
-#!/bin/bash
-# This script downloads the panel data that mykrobe uses for its predictions
```
Comment (bgruening) [inline recipes/mykrobe/post-link.sh]: why is this not needed anymore?

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

File: recipes/mykrobe/post-link.sh (hunk 1 above)
Comment (mbhall88) [reply recipes/mykrobe/post-link.sh]: Because the panel is no longer an external requirement of mykrobe. It is all handled internally now.

**Comment by mbhall88** [comment]
Hmm, testing this locally I am running into a problem in which mykrobe cannot find `mccortex31` where it expects.
```
INFO:mykrobe.cmds.amr:Running AMR prediction with panels data/panels/tb-species-170421.fasta.gz, data/panels/tb-hunt-probe-set-jan-03-2019.fasta.gz
Traceback (most recent call last):
  File "/home/vagrant/miniconda3/bin/mykrobe", line 11, in <module>
    sys.exit(main())
  File "/home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/cli.py", line 8, in main
    args.func(parser, args)
  File "/home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/parser.py", line 42, in run_subtool
    run(parser, args)
  File "/home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/cmds/amr.py", line 320, in run
    cp.run()
  File "/home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/typing/typer/genotyper.py", line 71, in run
    self._run_cortex()
  File "/home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/typing/typer/genotyper.py", line 86, in _run_cortex
    skeleton_dir=self.skeleton_dir,
  File "/home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/cortex/mccortex.py", line 141, in __init__
    super(McCortexGenoRunner, self).__init__()
  File "/home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/cortex/mccortex.py", line 21, in __init__
    raise RuntimeError(f"Did not find mccortex31. Expected it to be here: {self.mccortex31_path}. Cannot continue")
RuntimeError: Did not find mccortex31. Expected it to be here: /home/vagrant/miniconda3/lib/python3.7/site-packages/mykrobe/cortex/mccortex31. Cannot continue
```

**Comment by dpryan79** [comment]
Is mccortex31 a dependency or is that supposed to be part of the package?

**Comment by mbhall88** [comment]
> Is mccortex31 a dependency or is that supposed to be part of the package?

Both. It requires mccortex, but it requires a specific fork/commit that is specific to mykrobe's usage.  

I _think_ my latest commit should fix the problem.  

During the python setup.py installation it is supposed to copy the mccortex binary into a specific directory, but for some reason, the binary was not bein copied (and no error was being raised). I have no idea why this would happen. Moving the binary explicitly in the build scripts _seems_ to have fixed the problem from my tests locally.  
I will wait and test the CI artifacts though as a final step.

**Comment by mbhall88** [comment]
@bioconda-bot please fetch artifacts

**Comment by mbhall88** [comment]
Now it appears the panel data is not being included in the package.

```
INFO:mykrobe.cmds.amr:Running AMR prediction with panels data/panels/tb-species-170421.fasta.gz, data/panels/tb-hunt-probe-set-jan-03-2019.fasta.gz
Traceback (most recent call last):
  File "/home/vagrant/miniconda3/envs/mykrobe-test/bin/mykrobe", line 11, in <module>
    sys.exit(main())
  File "/home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/cli.py", line 8, in main
    args.func(parser, args)
  File "/home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/parser.py", line 42, in run_subtool
    run(parser, args)
  File "/home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/cmds/amr.py", line 320, in run
    cp.run()
  File "/home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/typing/typer/genotyper.py", line 71, in run
    self._run_cortex()
  File "/home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/typing/typer/genotyper.py", line 88, in _run_cortex
    self.mc_cortex_runner.run()
  File "/home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/cortex/mccortex.py", line 158, in run
    self._check_panels()
  File "/home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/cortex/mccortex.py", line 170, in _check_panels
    panel.filepath)
ValueError: Could not find a panel at /home/vagrant/miniconda3/envs/mykrobe-test/lib/python3.7/site-packages/mykrobe/data/panels/tb-species-170421.fasta.gz.
```

**Comment by mbhall88** [comment]
Ok, if I list all the files in this data directory during the bootstrap testing, I get the following list

```
15:31:39 BIOCONDA INFO (OUT) + find /opt/conda/conda-bld/mykrobe_1580138674478/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.6/site-packages/mykrobe/data -print
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/phylo
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/phylo/mtbc_hierarchy.json
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/predict
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/predict/tb
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/predict/tb/variant_to_resistance_drug.json
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/predict/taxon_coverage_threshold.json
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/predict/staph
15:31:39 BIOCONDA INFO (OUT) $PREFIX/lib/python3.6/site-packages/mykrobe/data/predict/staph/variant_to_resistance_drug.json
```
Whereas, if I print the list of files inside the same directory in my local installation of mykrobe I get
```
mykrobe/BX571856.1.fasta
mykrobe/phylo
mykrobe/phylo/mtbc_hierarchy.json
mykrobe/NC_000962.3.fasta
mykrobe/NC_000962.3.gb
mykrobe/predict
mykrobe/predict/tb
mykrobe/predict/tb/variant_to_resistance_drug.json
mykrobe/predict/tb/variant_to_resistance_drug-jan-03-2019.json
mykrobe/predict/taxon_coverage_threshold.json
mykrobe/predict/gn
mykrobe/predict/gn/variant_to_resistance_drug.json
mykrobe/predict/staph
mykrobe/predict/staph/variant_to_resistance_drug.json
mykrobe/tb-hierarchy.json
mykrobe/BX571856.1.gb
mykrobe/panels
mykrobe/panels/tb-amr-bradley_2015.fasta.gz
mykrobe/panels/tb-walker-probe-set-feb-02-2017.fasta.gz
mykrobe/panels/tb-bradley-probe-set-feb-09-2017.fasta.gz
mykrobe/panels/staph-amr-bradley_2015-feb-17-2017.fasta.gz
mykrobe/panels/tb-walker-probe-set-feb-09-2017.fasta.gz
mykrobe/panels/tb-walker-probe-set-jan-2019.fasta.gz
mykrobe/panels/tb-bradley-probe-set-jan-2019.fasta.gz
mykrobe/panels/tb-hunt-probe-set-jan-03-2019.fasta.gz
mykrobe/panels/staph-amr-probe_set_v0_3_13-160715.fasta.gz
mykrobe/panels/tb-bradley-probe-set-feb-02-2017.fasta.gz
mykrobe/panels/tb-k21-probe-set-feb-09-2017.fasta.gz
mykrobe/panels/tb-species-170421.fasta.gz
mykrobe/panels/staph-amr-probe_set_v0_3_13-160715.fasta
mykrobe/panels/tb-species-160330.fasta.gz
mykrobe/panels/staph-species-160227.fasta.gz
```
---
Adding this to the mccortex binary not being copied it looks like when conda installs/builds mykrobe for some reason the [two installation steps](https://github.com/Mykrobe-tools/mykrobe/blob/288890e770df742746a22403db7b0f3197f68e49/setup.py#L26-L28) (building mccortex and moving it's binary, and the download of the mykrobe panel data) are either not running, or the changes they make are not being maintained?

@dpryan79 are you able to provide any insight into what is going on here? I have been debugging this for the last couple of days and can't seem to get my head around what's going wrong.

**Comment by mbhall88** [comment]
Ok, this all seems very strange. When I change the install method from 
```
python -m pip install .
```
to
```
python setup.py install
```
All of the setup steps are run properly and all required files are copied/moved into the correct place.

**Comment by dpryan79** [comment]
Odd that `pip` is behaving so strangely.

**Comment by dpryan79** [comment]
@mbhall88 Go ahead and hit merge whenever you're ready with this.

**Comment by mbhall88** [comment]
@bioconda-bot please fetch artifacts

**Pull request merged**